pkg/model/clusterconfigentity.go

```diff
@@ -129,10 +130,16 @@ func newReconciliationSequence(cfg *ReconciliationSequenceConfig) *Reconciliatio
 	reconSeq.Queue = append(reconSeq.Queue, []*keb.Component{ //CRDs are always processed at the very beginning (or at the very end in deletion)
 		crdComponent,
 	})
-	cleanupComponent.Configuration = append(cleanupComponent.Configuration, keb.Configuration{Key: DeleteStrategyKey, Value: cfg.DeleteStrategy})
-	reconSeq.Queue = append(reconSeq.Queue, []*keb.Component{
-		cleanupComponent,
-	})
+
+	// if a cluster is pending deletion, we need to add the cleanup component into the reconciliation
+	if cfg.DesiredStatus.IsDeletion() {
```

Comment:
Shouldn 't we use here`IsDeleteCandidate`  ?

Reply:
I asked @clebs if we should use deleteCandidate and I think we decided for deletion. To be honest I don't quite remember why anymore. Could you jump in and help me here again?

Reply:
Did a change to accurately reflect the Deletion after discussion with @clebs , but this now requires a small upstream change in the CLI in order to not break it.